kernel/src/device/tty/termio.rs

```diff
@@ -57,57 +55,65 @@ bitflags! {
     }
 }
 
-impl Default for C_OFLAGS {
+impl Default for COutputFlags {
     fn default() -> Self {
-        C_OFLAGS::OPOST | C_OFLAGS::ONLCR
+        Self::OPOST | Self::ONLCR
     }
 }
 
+/// The control flags; `c_cflags` bits in Linux.
 #[repr(C)]
 #[derive(Debug, Clone, Copy, Pod)]
-pub struct C_CFLAGS(u32);
+pub(super) struct CCtrlFlags(u32);
 
-impl Default for C_CFLAGS {
+impl Default for CCtrlFlags {
     fn default() -> Self {
-        let cbaud = C_CFLAGS_BAUD::B38400 as u32;
-        let csize = C_CFLAGS_CSIZE::CS8 as u32;
-        let c_cflags = cbaud | csize | CREAD;
+        let cbaud = CCtrlBaud::B38400 as u32;
+        let csize = CCtrlSize::CS8 as u32;
+        let c_cflags = cbaud | csize | Self::READ_BIT;
         Self(c_cflags)
     }
 }
 
-impl C_CFLAGS {
-    pub fn cbaud(&self) -> Result<C_CFLAGS_BAUD> {
-        let cbaud = self.0 & CBAUD_MASK;
-        Ok(C_CFLAGS_BAUD::try_from(cbaud)?)
+impl CCtrlFlags {
+    const BAUD_MASK: u32 = 0x0000100f;
+    const SIZE_MASK: u32 = 0x00000030;
+    const READ_BIT: u32 = 0x00000080;
+
+    #[expect(dead_code)]
+    pub(super) fn baud(&self) -> Result<CCtrlBaud> {
+        let baud = self.0 & Self::BAUD_MASK;
+        Ok(CCtrlBaud::try_from(baud)?)
     }
 
-    pub fn csize(&self) -> Result<C_CFLAGS_CSIZE> {
-        let csize = self.0 & CSIZE_MASK;
-        Ok(C_CFLAGS_CSIZE::try_from(csize)?)
+    #[expect(dead_code)]
+    pub(super) fn size(&self) -> Result<CCtrlSize> {
+        let size = self.0 & Self::SIZE_MASK;
+        Ok(CCtrlSize::try_from(size)?)
     }
 
-    pub fn cread(&self) -> bool {
-        self.0 & CREAD != 0
+    #[expect(dead_code)]
+    pub(super) fn is_read(&self) -> bool {
+        self.0 & Self::READ_BIT != 0
     }
 }
 
-const CREAD: u32 = 0x00000080;
-const CBAUD_MASK: u32 = 0x0000100f;
-const CSIZE_MASK: u32 = 0x00000030;
-
+/// The size part of the control flags ([`CCtrlFlags`]).
 #[repr(u32)]
 #[derive(Clone, Copy, TryFromInt)]
-pub enum C_CFLAGS_CSIZE {
+pub(super) enum CCtrlSize {
+    // https://elixir.bootlin.com/linux/v6.0.9/source/include/uapi/asm-generic/termbits.h#L97
     CS5 = 0x00000000,
     CS6 = 0x00000010,
     CS7 = 0x00000020,
     CS8 = 0x00000030,
 }
 
+/// The baud part of the control flags ([`CCtrlFlags`]).
 #[repr(u32)]
 #[derive(Debug, Clone, Copy, TryFromInt)]
-pub enum C_CFLAGS_BAUD {
+pub(super) enum CCtrlBaud {
+    // https://elixir.bootlin.com/linux/v6.0.9/source/include/uapi/asm-generic/termbits-common.h#L30
     B0 = 0x00000000, /* hang up */
```

Comment:
It seems that for enums and bitflags, the Linux reference is added to the first variant, while for structs, it is above struct definition. Does this make any difference?

Reply:
Because there are multiple sources for `CInputFlags`:
https://github.com/asterinas/asterinas/blob/aa4b7aafd42e39fd6d89402bd8ab58e918cfe5b4/kernel/src/device/tty/termio.rs#L15-L31


So I'm currently enforcing this rule in this file, which avoids inconsistency among multiple enums/bitflags:
> for enums and bitflags, the Linux reference is added to the first variant